### ISSUE TITLE
API 1.0 features

### ISSUE THREAD

**Issue by ro0NL** [msgphp/symfony-demo-app#31]
Aiming to merge #28 as POC, thus reading items/collections. After that many topics remain open:

- [x] `GET /api/me` endpoint + JWT integration
- [x] leverage domain to document ID service
- [ ] content / access varying based on role
- [ ] add user resource fields like `{ roles: ['ROLE_X', ...], emails: ['x@y.z', ...] }`
- [ ] `GET /api/users/<id>/attributes` (not sure these need to be embedded)
- [x] auto re-index after domain writes (doctrine listener / event subscribers)
  - for being a "message driven" project we could investigate things like `dispatch(new SychronizeDomainObject($entity)`
- [ ] `POST /api/users/<id>/action` endpoints, i.e. `/confirm`, `/enable`, etc.
- [x] `POST /api/users` endpoint (registration)
- [x] `DELETE /api/users/<id>` endpoint
  - also write endpoints could simply dispatch our domain messages
- [ ] after #30: user avatars (file entities) could be embedded into `/api/users/<id>`
- [ ] investigate projection invalidation
  - given we have file & user resources; an updated file projection might invalidate a user projection, as the latter embeds the first.
- [ ] add filters
- [ ] add pagination

**Comment by prybus** [comment]
Hello @ro0NL,
when i use Jwt i have error:

> The service "security.authentication.provider.dao.api_login" has a dependency on a non-existent service "MsgPhp\User\Infra\Security\Jwt\SecurityUserProvider".

**Comment by ro0NL** [comment]
@prybus can you confirm the service is registered at https://github.com/msgphp/user-bundle/blob/master/Resources/config/security.php#L38? Otherwise if anything spiffy is going on.. please create a new issue :) Thx.
